shell/src/stats/memory.rs

```diff
@@ -0,0 +1,221 @@
+// Copyright (c) SimpleStaking and Tezedge Contributors
+// SPDX-License-Identifier: MIT
+
+use nix;
+use page_size;
+use serde::Serialize;
+use lazy_static::lazy_static;
+use regex::Regex;
+
+use std::error;
+use std::fmt;
+use std::fs::File;
+use std::io::prelude::*;
+use std::path::Path;
+use std::process::Command;
+use core::str::{from_utf8};
+
+pub type MemoryStatsResult<T> = std::result::Result<T, MemoryStatsError>;
+
+#[derive(Serialize, PartialEq, Clone, Debug)]
+pub struct LinuxData {
```

Comment:
Please provide description to structure fields. For example `dt` function is not obvious to a reader.